runtime/src/accounts_index.rs

```diff
@@ -750,30 +750,35 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
     ) -> Option<usize> {
         let mut current_max = 0;
         let mut rv = None;
-        for (i, (slot, _t)) in slice.iter().rev().enumerate() {
-            if *slot >= current_max && self.is_ancestor_or_root(*slot, ancestors, max_root) {
-                rv = Some((slice.len() - 1) - i);
-                current_max = *slot;
+        if let Some(ancestors) = ancestors {
+            if !ancestors.is_empty() {
+                for (i, (slot, _t)) in slice.iter().rev().enumerate() {
+                    if *slot >= current_max && ancestors.contains_key(slot) {
```

Comment:
@sakridge isn't *slot > current_max ok here? What conditions would cause = to be useful? If we already found a root or ancestor at a slot = to the current slot we're looking for, isn't that good enough? Do we expect multiple items in the slice to have the same slot #?

Reply:
`slot > current_max` should be ok as well. I don't believe there should be duplicates in the list.

Reply:
I worried about this... A problem is a test that uses slot 0. I thought this might be a corner case with unsigned and tests. -1 is such a convenient number but shockingly missing from unsigned types. ;-)

Reply:
I see, yea it should find slot=0 if it exists.